### PR TITLE
Milestone: API Integration 

### PR DESCRIPTION
**Pull request by NickEmma**
# Hi there 👋🏻

**Thanks for taking the time to review the project. Here are some things important to note about it:**

## Changes implemented in the "API Endpoint" branch

- ✅ Created an API endpoint to list all posts for a user.
- ✅ Created an API endpoint to list all comments for a user's post.
- ✅ Created an API endpoint to add a comment to a post. The owner of the comment is the user that makes it; it's not being set by passing a parameter.
- ✅ Added authentication to your API endpoints.
- ✅ API endpoints receive JSON and respond to JSON as well.

# Final notes 📔

_Is expected that this project meets the requirements. Please let us know if anything is left to be done or can be improved. Again, thanks for the review._

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,5 +1,11 @@
 Rails.application.routes.draw do
   devise_for :users, sign_out_via: [:get, :post]
```
Comment (danielufeli) [inline config/routes.rb:2]: - [ ] Nice working on your project, kindly check I am not able to sign up or sign in I get the below error when trying to sign up.
image

**Review by danielufeli — CHANGES_REQUESTED** [review]
Hi @Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Nice commits messages✔️
- Descriptive PR ✔️
- Good use of GitHub flow ✔️
- Clean Code✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @danielufeli in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (NickEmma) [reply config/routes.rb:2]: @danielufeli thank you for your review

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @NickEmma 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by NickEmma** [comment]
@thecodechaser thank you for approving my code😊

**Pull request merged**